master/buildbot/db/sourcestamps.py

```diff
@@ -188,3 +188,27 @@ def thd(conn):
 
             return ssdict
         return self.db.pool.do(thd)
+
+    def getSourceStampsForManyIds(self, sourcestamps_ids):
+        def thd(conn):
+            sourcestamps_tbl = self.db.model.sourcestamps
+
+            query = sa.select(columns=[sourcestamps_tbl.c.sourcestampsetid,
+                                       sourcestamps_tbl.c.branch,
+                                       sourcestamps_tbl.c.codebase,
+                                       sourcestamps_tbl.c.revision]). \
+                where(sourcestamps_tbl.c.sourcestampsetid.in_(sourcestamps_ids))
+
+            res = conn.execute(query)
+
+            sourcestamps = []
+            for row in res.fetchall():
+                sourcestamps.append({
+                    'sourcestampsetid': row.sourcestampsetid,
+                    'branch': row.branch,
+                    'codebase': row.codebase,
+                    'revision': row.revision,
+                    'short_revision': row.revision[:12],
```

Comment:
Why 12? Maybe we should move it to the configuration of the master? This twelve is a magic number.

Reply:
That is what Katana is using right now (13 characters for "short" revision).

I do agree on the magic thing, so it should either be a constant somewhere (I bet that this magic 12 appears in other places), or configurable as Kamil said.

I don't see any benefit in changing the number, so if we did move it to some config, it would still be the same 12. That being the case, I think that a constant is probably enough.